internal/proxy/connect_handler_test.go

```diff
@@ -42,203 +47,346 @@ func newConnHandlerTestDepsConfig(proxyEndpoint string) connHandlerTestDepsConfi
+	return httpConnHandleTestCase{
+		cfg:    httpTestDepsCfg,
+		node:   node,
+		server: server,
+		mux:    mux,
+	}
+}
+
+func teardownConnHandleHTTPTestCase(c httpConnHandleTestCase) {
+	defer func() { _ = c.node.Shutdown(context.Background()) }()
+	c.server.Close()
+}
+
+type connHandleTestCase struct {
+	connectProxyHandler *ConnectHandler
+	protocol            string
+}
+
+func newConnHandleTestCases(httpTestCase httpConnHandleTestCase, grpcTestCase grpcConnHandleTestCase) []connHandleTestCase {
+	return []connHandleTestCase{
+		{
+			connectProxyHandler: grpcTestCase.cfg.connectProxyHandler,
+			protocol:            "grpc",
+		},
+		{
+			connectProxyHandler: httpTestCase.cfg.connectProxyHandler,
+			protocol:            "http",
 		},
-		Data: []byte(customData),
 	}
+}
 
-	connReply, err := connHandler(context.Background(), testDepsCfg.connectEvent)
-	require.NoError(t, err)
-	require.Equal(t, connReply, expectedReply)
+func invokeTestConnectHandler(ctx context.Context, httpTestCase httpConnHandleTestCase, grpcTestCase grpcConnHandleTestCase, currentCase connHandleTestCase) (reply centrifuge.ConnectReply, err error) {
+	var connHandler centrifuge.ConnectingHandler
 
+	if currentCase.protocol == "http" {
+		connHandler = currentCase.connectProxyHandler.Handle(httpTestCase.node)
+		reply, err = connHandler(ctx, httpTestCase.cfg.connectEvent)
+	} else if currentCase.protocol == "grpc" {
+		connHandler = currentCase.connectProxyHandler.Handle(grpcTestCase.node)
+		reply, err = connHandler(ctx, grpcTestCase.cfg.connectEvent)
+	}
+
+	return reply, err
 }
 
-func TestHandleWithInvalidCustomData(t *testing.T) {
-	node := tools.NodeWithMemoryEngineNoHandlers()
-	defer func() { _ = node.Shutdown(context.Background()) }()
+func TestHandleConnectWithEmptyReply(t *testing.T) {
+	grpcTestCase := newConnHandleGRPCTestCase(proxyproto.NewProxyGRPCTestServer("", proxyproto.ProxyGRPCTestServerOptions{}))
+	defer teardownConnHandleGRPCTestCase(grpcTestCase)
 
-	mux := http.NewServeMux()
-	mux.HandleFunc("/proxy", func(w http.ResponseWriter, req *http.Request) {
-		_, _ = w.Write([]byte(`{"result": {"user": "56", "expire_at": 1565436268, "b64data": "invalid data"}}`))
+	httpTestCase := newConnHandleHTTPTestCase("/proxy")
+	httpTestCase.mux.HandleFunc("/proxy", func(w http.ResponseWriter, req *http.Request) {
+		_, _ = w.Write([]byte(`{}`))
 	})
-	server := httptest.NewServer(mux)
-	defer server.Close()
+	defer teardownConnHandleHTTPTestCase(httpTestCase)
 
-	testDepsCfg := newConnHandlerTestDepsConfig(server.URL + "/proxy")
-	connHandler := testDepsCfg.connectProxyHandler.Handle(node)
-	connReply, err := connHandler(context.Background(), testDepsCfg.connectEvent)
-	require.ErrorIs(t, err, centrifuge.ErrorInternal)
-	require.Equal(t, connReply, centrifuge.ConnectReply{})
+	cases := newConnHandleTestCases(httpTestCase, grpcTestCase)
+	for _, c := range cases {
+		reply, err := invokeTestConnectHandler(context.Background(), httpTestCase, grpcTestCase, c)
```

Comment:
is it possible to make invokeTestConnectHandler a method of connHandleTestCase?

Reply:
Good idea, done